framework/src/application/application.ts

```diff
@@ -156,38 +159,45 @@ export class Application {
 		genesisBlock: GenesisBlockJSON,
 		config: Partial<ApplicationConfig> = {},
 	) {
-		const errors = liskValidator.validate(genesisBlockSchema, genesisBlock);
+		const errors = validator.validate(genesisBlockSchema, genesisBlock);
 		if (errors.length) {
 			throw errors;
 		}
 		this._genesisBlock = genesisBlock;
 
 		// Don't change the object parameters provided
 		// eslint-disable-next-line no-param-reassign
-		config.rootPath = config.rootPath?.replace('~', os.homedir);
-		let appConfig = _.cloneDeep(config);
+		const appConfig = _.cloneDeep(applicationConfigSchema.default);
 
 		appConfig.label =
-			appConfig.label ??
+			config.label ??
 			`lisk-${this._genesisBlock.header.transactionRoot.slice(0, 7)}`;
 
-		appConfig = configurator.getConfig(appConfig, {
-			failOnInvalidArg: process.env.NODE_ENV !== 'test',
-		}) as ApplicationConfig;
+		mergeDeep(appConfig, config);
```

Comment:
If its lodahs merge, I would prefer to reassign values. 

```suggestion
appConfig = mergeDeep({}, appConfig, config);
```